server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -340,6 +350,24 @@ public ScheduledCancellable schedule(Runnable command, TimeValue delay, String e
         command = threadContext.preserveContext(command);
         if (Names.SAME.equals(executor) == false) {
             command = new ThreadedRunnable(command, executor(executor));
+        } else if (slowSchedulerWarnThresholdNanos > 0) {
+            final Runnable finalCommand = command;
+            command = () -> {
```

Comment:
This loses the `toString()` of the original command, which tends to cause debugging headaches. Not sure it matters much here today but let's avoid getting bitten by that in the future.